### PR TITLE
Vehicle Spawner + Fix 3 Bugs that Caused the Menu to be Unusable

### LINKED ISSUE #76: Crash on load
I am loading the menu like normal, on the main branch. and then it just gives this stack trace. I did compile this on my pc, with debug mode on, but as I said it was a fresh branch.
[stacktrace.txt](https://github.com/YimMenu/HorseMenu/files/14440186/stacktrace.txt)

### PR DESCRIPTION
**Pull request by Rxann**
Things I changed:
Fixed the crash that was caused by the folder for settings being in Documents, so it is now in appdata.
Added a mutex to notifications to fix a crash that was caused.
Fixed/removed debug menu, so there is no longer a crash when clicking into the debug menu
Added a vehicle spawner function, and then added the interface for spawning a bounty wagon for you or another user.

Let me know if there are any code issues, but functionally it works well.

Fixes #76 
Possible fix for #48 but we will see

### CODE REVIEW

**Comment by DayibBaba** [comment]
Why did you feel like appdata would be better suited to host the files?

**Comment by Rxann** [comment]
> Why did you feel like appdata would be better suited to host the files?

When using documents, the menu would crash on load and not save the files, etc.

I don't know if this was just a me issue, I wouldn't be surprised if it was but YimMenu also uses appdata and I think its a generally good place.

**Comment by Hyman-R** [comment]
Please update the binary release so that we can share the fixed version, thanks.

File: src/util/VehicleSpawner.hpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include "game/backend/ScriptMgr.hpp"
+#include "game/features/Features.hpp"
+#include "game/rdr/Natives.hpp"
+
+#include <cmath>
+namespace YimMenu
+{
+	inline bool SpawnVehicle(std::string model_name, Ped player_ped_id);
```
Comment (MarkEcza) [inline src/util/VehicleSpawner.hpp:9]: Could use extern here

File: src/util/VehicleSpawner.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#include "VehicleSpawner.hpp"
+
+namespace YimMenu
+{
+	inline bool SpawnVehicle(std::string model_name, Ped player_ped_id)
```
Comment (MarkEcza) [inline src/util/VehicleSpawner.cpp:5]: And removing inline here

**Comment by Rxann** [comment]
@MarkEcza thanks for the review, i will have these things fixed in like 7 hours 🤣

File: src/main.cpp [hunk 3]
```diff
@@ -1,24 +1,25 @@
...
+#include "game/pointers/Pointers.hpp"
+
 
 namespace YimMenu
 {
 	DWORD Main(void*)
 	{
-		const auto documents = std::filesystem::path(std::getenv("USERPROFILE")) / "Documents";
-		FileMgr::Init(documents / "HellBase"); // TODO
+		const auto documents = std::filesystem::path(std::getenv("appdata")) /= "HorseMenu";
```
Comment (Yimura) [inline src/main.cpp:21]: Heh, why is there a `/=` in the middle of it? Seems like UB, I'm specifically talking about `=` coming after the `/`.

**Review by Yimura — CHANGES_REQUESTED** [review]

File: src/main.cpp (hunk 3 above)
Comment (Rxann) [reply src/main.cpp:21]: What does UB mean? i'm a little out of touch with some programmer slang. The /= does the same thing as just /

**Comment by Rxann** [comment]
@Yimura do you want me to change it back to the `documents / "HorseMenu"` in the Filemgr call?

**Comment by Yimura** [comment]
> @Yimura do you want me to change it back to the `documents / "HorseMenu"` in the Filemgr call?

I don't really care about the save location really, I do care about the `/=` weirdness. I'd just use `/` instead.

**Comment by Rxann** [comment]
> > @Yimura do you want me to change it back to the `documents / "HorseMenu"` in the Filemgr call?
> 
> 
> 
> I don't really care about the save location really, I do care about the `/=` weirdness. I'd just use `/` instead.

alright it has been fixed.

**Pull request merged**